### PR TITLE
Refactor End to End tests

### PR DESCRIPTION
**Pull request by mattonfoot**
# Context

Refactors cypress suite and adds form submission test example

## Changes proposed in this pull request

- Refactor page objects to align more with template inheritance
- Refactor common UI components into separate component objects
- include an example form fill action on device wearer page
- incorporate AXE Accessibility testing feature

### CODE REVIEW

File: server/models/DeviceWearer.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ const DeviceWearerModel = z.object({
   lastName: z.string().nullable(),
   alias: z.string().nullable(),
   dateOfBirth: z.string().datetime().nullable(),
-  adultAtTimeOfInstallation: z.boolean().nullable(),
+  adultAtTimeOfInstallation: z.coerce.boolean().nullable(),
```
Comment (mattonfoot) [inline server/models/DeviceWearer.ts:12]: This is probably an important change to note, the wiremock response returns boolean values as a string and so zod needs to coerce strings into booleans to be successful.

File: integration_tests/mockApis/cemo.ts [hunk 2]
```diff
@@ -125,6 +125,81 @@ const getOrder = (options: GetOrderStubOptions = defaultGetOrderOptions): SuperA
...
+      urlPattern: `/cemo/api/order/${options.id}/device-wearer`,
+    },
+    response: {
+      status: options.httpStatus,
+      headers: { 'Content-Type': 'application/json;charset=UTF-8' },
+      body: options.httpStatus === 200 ? '{{{request.body}}}' : null,
+      transformers: ['response-template'],
+    },
+  })
+
```
Comment (ben-cart3r) [inline integration_tests/mockApis/cemo.ts:202]: I think the naming may be wrong, this endpoint isn't posting the entirety of the order, just the device-wearer

File: server/models/DeviceWearer.ts (hunk 1 above)
Comment (ben-cart3r) [reply server/models/DeviceWearer.ts:12]: This model is only used to validate the response from the api which should only return boolean values and anything other would be incorrect. Also `.zod.coerce.boolean()` uses the `Boolean` primitive which in JS makes anything not falsy into true, e.g.

```node
> Boolean('abc')
true
```

File: integration_tests/mockApis/cemo.ts (hunk 2 above)
Comment (mattonfoot) [reply integration_tests/mockApis/cemo.ts]: This is renamed more appropriately now

File: server/models/DeviceWearer.ts (hunk 1 above)
Comment (mattonfoot) [reply server/models/DeviceWearer.ts:12]: Created a definable response so this is not required anymore

**Review by ben-cart3r — APPROVED** [review]
👍

**Pull request merged**